lib/action_view/component/base.rb

```diff
@@ -115,21 +115,21 @@ def has_initializer?
         end
 
         def source_location
-          # Require #initialize to be defined so that we can use
-          # method#source_location to look up the file name
-          # of the component.
-          #
-          # If we were able to only support Ruby 2.7+,
-          # We could just use Module#const_source_location,
-          # rendering this unnecessary.
-          raise NotImplementedError.new("#{self} must implement #initialize.") unless has_initializer?
-
-          instance_method(:initialize).source_location[0]
+          if RUBY_VERSION < "2.7"
```

Comment:
it would be much cleaner to test if the method we need exits. How about we test `if Module.method_defined(:const_source_location)`?

Reply:
I had gone back and forth on this, but settled on using RUBY_VERSION, as it might provide a better clue to delete the block when we drop support for rubies < 2.7

But a comment here like `# to support Ruby < 2.7` would also do the trick.

Reply:
Yeah I think comments would fix it. I'd be worried that in Ruby 2.8 they rename `const_source_location` to something else, and we'd break in some harder to understand way.

Reply:
What if we checked to see if the component responds_to `const_source_location`?